libvast/vast/detail/posix.hpp

```diff
@@ -18,6 +22,27 @@ namespace vast::detail {
 
 enum class socket_type { datagram, stream, fd };
 
+/// Holds the necessary state to send unix datagrams to a destination socket.
+struct uds_datagram_sender {
+  /// The destructor unlinks the src socket.
+  ~uds_datagram_sender();
+
+  /// Helper function to create a sender.
+  static caf::expected<uds_datagram_sender> make(const std::string& path);
+
+  /// Send the content of `data to `dst`.
+  caf::error send(span<char> data);
```

Comment:
Should be `span<const std::byte>` to be in line with our other code.

Reply:
The underlying API works with char, I don't see a reason to bring unnecessary casting upon ourselves.